### PR TITLE
Add additional quick mode test cases

### PR DESCRIPTION
**Pull request by JukkaL**
Some of these are currently skipped because they are crashing.

Add more descriptive names to existing --quick test cases.
Also minor tweaks to existing test cases.

### CODE REVIEW

File: test-data/unit/check-incremental.test [hunk 1]
```diff
@@ -2312,3 +2316,286 @@ T = TypeVar('T')
...
+[case testQuickAndDirtyTurnGenericClassIntoNonGeneric-skip]
+# flags: --quick-and-dirty
+import a
+[file a.py]
+import b
+def f() -> b.C[int]: pass
+[file b.py]
+from typing import TypeVar, Generic
+import a
+
```
Comment (gvanrossum) [inline test-data/unit/check-incremental.test]: No need for blank lines in this file IMO. (Same in the next test.)

File: test-data/unit/check-incremental.test [hunk 2]
```diff
@@ -2312,3 +2316,286 @@ T = TypeVar('T')
...
+[case testQuickAndDirtyRenameModule]
+# flags: --quick-and-dirty
+import a
+[file a.py]
+import b
+b.f()
+[file b.py]
+def f() -> None: pass
+[delete b.py.2]
+[file c.py]
```
Comment (gvanrossum) [inline test-data/unit/check-incremental.test]: Should this be c.py.2?

File: test-data/unit/check-incremental.test [hunk 3]
```diff
@@ -2285,7 +2289,7 @@ S = str
 import b
 def f(x: b.S) -> int: return 0
```
Comment (gvanrossum) [inline test-data/unit/check-incremental.test]: The remaining tests don't have [rechecked] and [stale] fields -- not sure if this is intentional?

File: test-data/unit/check-incremental.test [hunk 4]
```diff
@@ -2112,7 +2116,7 @@ tmp/a.py:2: error: Incompatible return value type (got "str", expected "int")
 [rechecked a]
 [stale]
 
-[case testQuickAndDirty6]
+[case testQuickAndDirtyExistingError]
```
Comment (JukkaL) [inline test-data/unit/check-incremental.test]: Updated

File: test-data/unit/check-incremental.test [hunk 5]
```diff
@@ -2130,7 +2134,7 @@ tmp/a.py:2: error: Incompatible return value type (got "float", expected "int")
 [rechecked a]
 [stale]
 
-[case testQuickAndDirty7]
+[case testQuickAndDirtyIntroduceReferencesWithinCycle]
```
Comment (JukkaL) [inline test-data/unit/check-incremental.test]: Added two additional test cases

File: test-data/unit/check-incremental.test (hunk 3 above)
Comment (JukkaL) [inline test-data/unit/check-incremental.test]: Yes, they seemed to not add value when multiple test cases are very similar. I double checked and added those when they actually seemed useful.

File: test-data/unit/check-incremental.test (hunk 1 above)
Comment (JukkaL) [inline test-data/unit/check-incremental.test]: Updated

File: test-data/unit/check-incremental.test (hunk 2 above)
Comment (JukkaL) [inline test-data/unit/check-incremental.test]: Yes -- good catch

**Pull request merged**